packages/inspector/src/components/EntityInspector/ScriptInspector/ScriptInspector.tsx

```diff
@@ -0,0 +1,427 @@
+import MoreOptionsMenu from '../MoreOptionsMenu';
+import { Button } from '../../Button';
+import { ScriptParamField } from './ScriptParamField';
+import { CreateScriptModal } from './CreateScriptModal';
+
+import { getScriptTemplateClass } from './templates';
+import {
+  isValidPath,
+  parseLayout,
+  isScriptNode,
+  isScriptNameAvailable,
+  buildScriptPath,
+  readScript,
+  mergeLayout,
+} from './utils';
+import { getScriptParams } from './parser';
+import type { Props, ScriptLayout, ScriptParamUnion, ChangeEvt, ScriptItem } from './types';
+
+import './ScriptInspector.css';
+
+export default withSdk<Props>(({ sdk, entity: entityId, initialOpen = true }) => {
+  const { Script } = sdk.components;
+  const dispatch = useAppDispatch();
+  const files = useAppSelector(selectAssetCatalog);
+
+  const hasScript = useHasComponent(entityId, Script);
+  const [componentValue, setComponentValue] = useComponentValue(entityId, Script);
+  const [emptyScriptModuleMode, setEmptyScriptModuleMode] = useState<
+    'create' | 'import' | undefined
+  >(undefined);
+  const [showCreateModal, setShowCreateModal] = useState(false);
+  const [error, setError] = useState<string | undefined>(undefined);
+  const scripts = componentValue?.value ?? [];
+
+  const addScript = useCallback(
+    (script: ScriptItem) => {
+      setComponentValue({ value: [...scripts, script] });
+    },
+    [scripts, setComponentValue],
+  );
+
+  const updateScript = useCallback(
+    (index: number, script: ScriptItem) => {
+      const newScripts = [...scripts];
+      newScripts[index] = script;
+      setComponentValue({ value: newScripts });
+    },
+    [scripts, setComponentValue],
+  );
+
+  const removeScript = useCallback(
+    (index: number) => {
+      const newScripts = scripts.filter((_, i) => i !== index);
+      setComponentValue({ value: newScripts });
+    },
+    [scripts, setComponentValue],
+  );
+
+  // memoize parsed layouts to avoid re-parsing on every render
+  const parsedLayouts = useMemo(() => {
+    return scripts.map(script => parseLayout(script.layout));
+  }, [scripts]);
+
+  const createScript = useCallback(
+    (path: string, priority = 0, content: string) => {
+      const { params, error } = getScriptParams(content);
+      const layout: ScriptLayout = { params, error };
+
+      const newScript: ScriptItem = {
+        path,
+        priority,
+        layout: JSON.stringify(layout),
+      };
+
+      addScript(newScript);
+      setEmptyScriptModuleMode(undefined);
+    },
+    [addScript, setEmptyScriptModuleMode],
+  );
+
+  const handleRemove = useCallback(async () => {
+    sdk.operations.removeComponent(entityId, Script);
+    await sdk.operations.dispatch();
+  }, [sdk, entityId, Script]);
+
+  const handleRemoveScript = useCallback(
+    (index: number) => {
+      removeScript(index);
+    },
+    [removeScript],
+  );
+
+  const handleEditScript = useCallback(
+    async (index: number) => {
```

Comment:
we should pass here the `event` param and do a `event.preventDefault();` to avoid collapsing/expanding the `<Container />` component